elyra/pipeline/validation.py

```diff
@@ -697,6 +722,33 @@ def _is_required_property(self, property_dict: dict, node_property: str) -> bool
         """
         node_op_parameter_list = property_dict['uihints']['parameter_info']
         for parameter in node_op_parameter_list:
-            if parameter['parameter_ref'] == node_property:
+            if parameter['parameter_ref'] == f"elyra_{node_property}":
                 return parameter['data']['required']
         return False
+
+    def _get_component_type(self, property_dict: dict, node_property: str) -> str:
+        """
+        Helper function to determine the type of a node property
+        :param property_dict: a dictionary containing the full list of property parameters and descriptions
+        :param node_property: the property to look for
+        :return: the data type associated with node_property, defaults to 'string'
+        """
+        for prop in property_dict['uihints']['parameter_info']:
+            if prop["parameter_ref"] == f"elyra_{node_property}":
+                return prop['data'].get('format', 'string')
+
+    def _get_parent_id_list(self, pipeline_definition, node_id_list: list, parent_list: list) -> List:
+        """
+        Helper function to return a complete list of parent node_ids
+        :param pipeline_definition: the complete pipeline definition
+        :param node_id_list: list of parent node ids
+        :param parent_list: the list to add additional found parent node ids
+        :return:
+        """
+        for node_id in node_id_list:
+            node = pipeline_definition.get_node(node_id)
+            if node:
+                parent_list.append(node_id)
+                node_ids = list(x.get('node_id_ref', None) for x in node.get_component_links)
+                self._get_parent_id_list(pipeline_definition, node_ids, parent_list)
+        return parent_list
```

Comment:
I haven't tested this, but it looks like this validation comes before the circular reference check. So we should make sure this doesn't end up in an infinite loop. Also, not sure if this handles subflows?